src/v0/destinations/am/transform.js

```diff
@@ -79,112 +80,138 @@ const aliasEndpoint = (destConfig) => {
   return retVal;
 };
 
-function handleSessionIdUnderRoot(message) {
-  const sessionId = get(message, 'session_id');
+const handleSessionIdUnderRoot = (sessionId) => {
   if (typeof sessionId === 'string') {
-    return sessionId.substr(sessionId.lastIndexOf(':') + 1, sessionId.length);
+    const extractedPart = sessionId.split(':').reverse();
+    if (!isValidInteger(extractedPart[0])) return -1;
+    return Number(extractedPart[0]);
   }
-  return sessionId;
-}
+  return Number(sessionId);
+};
 
-function handleSessionIdUnderContext(message) {
-  let sessionId = get(message, 'context.sessionId');
-  sessionId = Number(sessionId);
-  if (Number.isNaN(sessionId)) return -1;
-  return sessionId;
-}
+const handleSessionIdUnderContext = (sessionId) => {
+  if (!isValidInteger(sessionId)) return -1;
+  return Number(sessionId);
+};
 
-function getSessionId(message) {
-  return get(message, 'session_id')
-    ? handleSessionIdUnderRoot(message)
-    : get(message, 'context.sessionId')
-    ? handleSessionIdUnderContext(message)
-    : -1;
-}
+const checkForJSONAndUserIdLengthAndDeviceId = (jsonBody, userId, deviceId) =>
+  Object.keys(jsonBody).length === 0 ||
+  (userId &&
+    userId.length < 5 &&
+    (!batchEventsWithUserIdLengthLowerThanFive ||
+      (batchEventsWithUserIdLengthLowerThanFive && !deviceId)));
+const getSessionId = (message) => {
+  let sessionId = -1;
+  const rootSessionId = get(message, 'session_id');
+  if (rootSessionId) {
+    sessionId = handleSessionIdUnderRoot(rootSessionId);
+    if (sessionId !== -1) {
+      return sessionId;
+    }
+  }
+  const contextSessionId = get(message, 'context.sessionId');
+  if (contextSessionId) {
+    sessionId = handleSessionIdUnderContext(contextSessionId);
+  }
+  return sessionId;
+};
 
-function addMinIdlength() {
-  return { min_id_length: 1 };
-}
+const addMinIdlength = () => ({ min_id_length: 1 });
 
-function setPriceQuanityInPayload(message, rawPayload) {
+const setPriceQuanityInPayload = (message, rawPayload) => {
   let price;
   let quantity;
-  if (isDefinedAndNotNull(message.properties.price)) {
+  if (isDefinedAndNotNull(message.properties?.price)) {
     price = message.properties.price;
-    quantity = message.properties.quantity || 1;
+    quantity = message.properties?.quantity || 1;
   } else {
-    price = message.properties.revenue;
+    price = message.properties?.revenue;
     quantity = 1;
   }
   rawPayload.price = price;
   rawPayload.quantity = quantity;
-  rawPayload.revenue = message.properties.revenue;
+  rawPayload.revenue = message.properties?.revenue;
```

Comment:
[Q] If `properties.revenue` is not present, `revenue` value would be `undefined`, will it rightly put data into Amplitude ? 🤔

Reply:
Should it default to valid value which indicates nothing like may be `0` or `0.0` ?

Reply:
Yess we won't be calculating revenue from our side for this one as already discussed its a customer expected property and they can easily use user transformation for this.
Reasons we are not doing it at our end:
1. Not much destinations do this
2. breaking change
3. Easily achievable through ut